README.md

```diff
@@ -1 +1,588 @@
+
+If we go back to our code snippet, we can see that the following happens:
+
+- `console.log(1);` is called, and logged to the console
+- We schedule a new _microtask_ with `Promise.resolve().then(() => {})`, but at this point in time, JavaScript has not finished executing; it's not time to actually process microtasks just yet
+- JavaScript continues executing code, because we're not done yet
+- `console.log(3);` is called, and logged to the console
+- JavaScript has finished executing! The browser can now process any microtasks we may have
+- `console.log(2);` is called, and logged to the console.
+
+So why is this important? How does this allow us to use microtasks to _batch_ work? Let's take a look at a practical example:
+
+```js
+/* We declare and initialize a variable to keep track of whether or not an update has already been requested */
+let updateRequested = false;
+
+function scheduleUpdate() {
+  /** 
+   * In here, we need a check to see if an update is already previously requested.
+   * If an update already is requested, we don't want to do any unnecessary work!
+   */
+  if(!updateRequested) {
+
+    /* If no update has yet been requested, we set the `updateRequested` flag to `true` */
+    updateRequested = true;
+
+    /** 
+     * Since we now know that microtasks run after JavaScript has finished
+     * executing, we can use this knowledge to our advantage, and only set 
+     * the `updateRequested` flag back to `false` again once all the tasks 
+     * have run, essentially delaying, or _batching_ the update!
+     */
+    Promise.resolve().then(() => {
+      updateRequested = false;
+      update();
+    });
+  }
+}
+
+/* This is our `update` function that we only want to be called once */
+function update() {
+  console.log('updating!');
+}
+
+/* We call scheduleUpdate in quick succession */
+scheduleUpdate();
+scheduleUpdate();
+scheduleUpdate();
+
+// "updating!" only logged once
+```
+
+What we do here is essentially _gatekeep_ potentially multiple incoming calls for `scheduleUpdate` with the `if` statement, and cleverly use our newfound knowledge of microtasks to our advantage to process any other incoming calls _first_, and only once JavaScript has finished executing, and when its microtask processing time, our `update` function is called.
+
+As a fun tidbit of knowledge, we can also write the `scheduleUpdate` function like so:
+
+```js
+function scheduleUpdate() {
+  if(!updateRequested) {
+    updateRequested = true;
+    updateRequested = await false;
+    update();
+  }
+}
+```
+
+Since, from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await):
+> If the value of the expression following the await operator is not a Promise, it's converted to a resolved Promise.
```

Comment:
```suggestion
> If the value of the expression following the await operator is not a Promise, it's converted to a resolved Promise.

In otherwords, once we set `updateRequested`, we don't immediately unset, the `await` keyword ensures that we only _schedule_ it to be unset. 
```

Reply:
Something i've been wondering is whether  each `await` keyword schedules a new microtask, and what the ramifications of the answer are. maybe worth a footnote